src/reconcile/rec_visibility.c

```diff
@@ -19,12 +19,15 @@ __rec_update_save(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, WT_
     WT_SAVE_UPD *supd;
 
     /* If nothing is committed, we must restore the update chain. */
+    // WT-9417 IGNORE
     WT_ASSERT(session, onpage_upd != NULL || supd_restore);
```

Comment:
These asserts are also very important, so I would strong both of them.

Reply:
Updated in [06a51d4](https://github.com/wiredtiger/wiredtiger/pull/8004/commits/06a51d4524e468184a6ca8e26a3c9f78472d7910)